### ISSUE TITLE
Random (rare) crash during synchronisation attempt

### ISSUE THREAD

**Issue by simonspa** [stefan-niedermann/nextcloud-deck#475]
**Describe the bug**
Sometimes I manage to run into random synchronisation errors such as the stack trace appended below. Server is working fine and does not show any errors in the log (neither NC nor webserver).

**To Reproduce**
1. Go to aboard
2. Swipe lists back and forth
3. Pull down to sync in probably too quick succession?

**Additional information**
Sometimes github bug reports can paint a skewed picture because they only present problems but never a laudation. Therefore, let me include as additional information that I find this app really great. It works (almost) flawlessly, has a great UX and really boosts my productivity. Thank you very much for investing so much time in writing this high-quality software!

**Stacktrace**

```
App Version: 1.2.1
App Version Code: 1002001
App Flavor: play

Files App Version Code: 30110190

---

OS Version: 3.4.0+(e730b2f661)
OS API Level: 25
Device: onyx
Manufacturer: OnePlus
Model (and Product): ONE E1003 (onyx)

---

java.lang.NullPointerException: Attempt to invoke interface method 'void it.niedermann.nextcloud.deck.model.interfaces.IRemoteEntity.setAccountId(long)' on a null object reference
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.SyncHelper$1.onResponse(SyncHelper.java:40)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.SyncHelper$1.onResponse(SyncHelper.java:34)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.providers.CardDataProvider$1.onResponse(CardDataProvider.java:52)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.providers.CardDataProvider$1.onResponse(CardDataProvider.java:47)
	at it.niedermann.nextcloud.deck.api.RequestHelper$ResponseConsumer.accept(RequestHelper.java:56)
	at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeOnObserver.onNext(ObservableSubscribeOn.java:58)
	at io.reactivex.internal.operators.observable.ObservableFromPublisher$PublisherSubscriber.onNext(ObservableFromPublisher.java:56)
	at com.nextcloud.android.sso.api.NextcloudAPI.lambda$performRequestObservable$0$NextcloudAPI(NextcloudAPI.java:86)
	at com.nextcloud.android.sso.api.-$$Lambda$NextcloudAPI$yhhE-DMCe81fXF-bhjup5Tra-6o.subscribe(lambda)
	at io.reactivex.internal.operators.observable.ObservableFromPublisher.subscribeActual(ObservableFromPublisher.java:31)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:272)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
	at java.lang.Thread.run(Thread.java:761)
```

**Comment by stefan-niedermann** [comment]
Should be fixed after solving https://github.com/stefan-niedermann/nextcloud-deck/issues/554

Please watch this issue to get noticed in which version this will be implemented. If it still happens after a new version containing the fix of #554 appeared, please open a new issue.

Unfortunately the issue itself is pretty hard to track down and we are not able to fix it based on the given information, sorry.

**Comment by stefan-niedermann** [comment]
> Sometimes github bug reports can paint a skewed picture because they only present problems but never a laudation. Therefore, let me include as additional information that I find this app really great. It works (almost) flawlessly, has a great UX and really boosts my productivity. Thank you very much for investing so much time in writing this high-quality software!

Made me smile btw :slightly_smiling_face:
